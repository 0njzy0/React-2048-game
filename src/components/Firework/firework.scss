:global {

  $particles: 50;
  $width: 500;
  $height: 500;

  // Create the explosion...
  $box-shadow: ();
  $box-shadow2: ();
  @for $i from 0 through $particles {
    $box-shadow: $box-shadow,
      random($width)-$width / 2 + px
      random($height)-$height / 1.2 + px
      hsl(random(360), 100, 50);
    $box-shadow2: $box-shadow2, 0 0 #fff;
  }

  @mixin animation-delay ($settings) {
    animation-delay: $settings;
  }

  @mixin animation-duration ($settings) {
    animation-duration: $settings;
  }

  @mixin animation ($settings) {
    animation: $settings;
  }

  @mixin transform ($settings) {
    transform: $settings;
  }

  .pyro > .before,
  .pyro > .after {
    border-radius: 50%;
    box-shadow: $box-shadow2;
    height: 5px;
    position: absolute;
    width: 5px;
    @include animation((1s bang ease-out infinite backwards, 1s gravity ease-in infinite backwards, 5s position linear infinite backwards));
  }

  .pyro > .after {
    @include animation-delay((1.25s, 1.25s, 1.25s));
    @include animation-duration((1.25s, 1.25s, 6.25s));
  }

  /* stylelint-disable max-nesting-depth */
  @keyframes :global(bang) {
    to {
      box-shadow: $box-shadow;
    }
  }

  @keyframes :global(gravity) {
    to {
      @include transform(translateY(200px));
      opacity: 0;
    }
  }

  @keyframes :global(position) {
    0%,
    19.9% {
      margin-left: 40%;
      margin-top: 10%;
    }

    20%,
    39.9% {
      margin-left: 30%;
      margin-top: 40%;
    }

    40%,
    59.9% {
      margin-left: 70%;
      margin-top: 20%;
    }

    60%,
    79.9% {
      margin-left: 20%;
      margin-top: 30%;
    }

    80%,
    99.9% {
      margin-left: 80%;
      margin-top: 30%;
    }
  }
}
