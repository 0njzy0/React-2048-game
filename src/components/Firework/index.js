import React from "react";
// import PropTypes from 'prop-types';
import "./firework.scss";

export default function Firework() {
  return (
    <div className="firework">
      <div className="before" />
      <div className="after" />
    </div>
  );
}
