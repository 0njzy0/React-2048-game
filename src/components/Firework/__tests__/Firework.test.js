import React from 'react';
import renderer from 'react-test-renderer';
import Firework from '..';

describe('<Firework />', () => {
  it('component render', () => {
    const firework = renderer.create(
      <Firework />,
    ).toJSON();
    expect(firework).toMatchSnapshot();
  });
});
