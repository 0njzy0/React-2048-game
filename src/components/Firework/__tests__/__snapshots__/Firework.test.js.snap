// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Firework /> component render 1`] = `
<div
  className="pyro"
>
  <div
    className="before"
  />
  <div
    className="after"
  />
</div>
`;
